### ISSUE TITLE
client async reading of lines does not stop when connection closed serverside

### ISSUE THREAD

**Issue by heeen** [fafhrd91/aiohttp#6007]
### Describe the bug

when reading content from a POST line by line, the loop is not broken by a connection closed exception or otherwise.

### To Reproduce

implement a long running http endpoint such as a SSE connection

then, in a client read the content as such:

```python
    headers = {"SimID": str(self.simId)}
    timeout = aiohttp.ClientTimeout()  # no timeout
     self.clientSession = aiohttp.ClientSession(
            raise_for_status=True, headers=headers, timeout=timeout)
        self.connection = await self.clientSession.post("server_host/v1/sse",
                                                            json=connect_data})
    
        try:
            async for line in self.connection.content:
                print(
                    f"yield {line} closed? {self.connection.closed} {self.connection.content.at_eof()} {self.connection.content.is_eof()}")
                yield line
        except Exception as e:
            print(f"Exception: {type(e)} {e.args} {e}")
```

the loop is never exited, is_eof(), at_eof() and connection.closed is False for the last line read.

in a second task, connection.closed is True however.

### Expected behavior

if the client closes the connection from a separate task, a ClientConnectionError is risen from which the loop exits, the same should be expected if the server closes the connection

### Logs/tracebacks

```python-traceback
yield: b'event:clusters:connect\n' closed? False False False
```

### Python Version

```console
$ python --version
Python 3.9.5
```

### aiohttp Version

```console
$ python -m pip show aiohttp
Name: aiohttp
Version: 3.7.4.post0
Summary: Async http client/server framework (asyncio)
Home-page: https://github.com/aio-libs/aiohttp
Author: Nikolay Kim
Author-email: fafhrd91@gmail.com
License: Apache 2
Location: /home/florian/.local/lib/python3.9/site-packages
Requires: chardet, typing-extensions, multidict, async-timeout, attrs, yarl
Required-by:
```

### multidict Version

```console
$ python -m pip show multidict
```

### yarl Version

```console
$ python -m pip show yarl
```

### OS

Linux

### Related component

Client

### Code of Conduct

- [X] I agree to follow the aio-libs Code of Conduct

**Comment by heeen** [comment]
if you look at connection.content after the server closed the connection, there is no _eof_waiter and no _waiter - shouldn't there be one if I async loop over lines?

content.set_exception manually does not work

connection.close() or .release() also has no effect if the connection is already closed.

**Comment by achimnol** [comment]
I confirm the same symptom with SSE (server-sent events) subscription handler implemented on top of aiohttp's client-side API: the client is not interrupted when the server shuts down and closes all connections.

My env: Python 3.9.7 and aiohttp 3.7.4.post0 (both server and client side) + aiohttp-sse 2.0.0 (server-side) on Linux (Ubuntu 20.04) amd64.
ps) Testing the same code with aiohttp 3.8.0 also has the same issue.

**Comment by Dreamsorcerer** [comment]
Appears to be working for me today:
client:
```
import asyncio
import aiohttp

async def main():
    timeout = aiohttp.ClientTimeout()  # no timeout
    async with aiohttp.ClientSession(timeout=timeout) as sess:
        async with sess.get("http://localhost:8080/hello") as resp:
            try:
                async for line in resp.content:
                    print(line)
            except Exception as e:
                print(f"Exception: {type(e)} {e.args} {e}")
            print(resp.closed, resp.content.at_eof(), resp.content.is_eof())

asyncio.run(main())
```

server:
```
import asyncio
import json
from datetime import datetime
from aiohttp import web
from aiohttp_sse import sse_response

async def hello(request: web.Request) -> web.StreamResponse:
    async with sse_response(request) as resp:
        while resp.is_connected():
            time_dict = {"time": f"Server Time : {datetime.now()}"}
            data = json.dumps(time_dict, indent=2)
            print(data)
            await resp.send(data)
            await asyncio.sleep(1)
            break
    return resp

app = web.Application()
app.router.add_route("GET", "/hello", hello)
web.run_app(app, host="127.0.0.1", port=8080)
```
